meson.build

```diff
@@ -0,0 +1,60 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+project('qrtr',
+        'c',
+        license : [ 'BSD-3-Clause'],
+        meson_version : '>= 0.64.0', # for feature user options
+        default_options : [
+        #     'warning_level=2',	# sets -Wextra
+        'buildtype=release',
+        ])
+
+# Set advanced compiler flags
+compiler = meson.get_compiler('c')
+
+prefix = get_option('prefix')
+systemd = dependency('systemd')
+systemd_system_unit_dir = systemd.get_variable(
+        pkgconfig : 'systemdsystemunitdir',
+        pkgconfig_define: ['prefix', prefix])
+
+compiler_cflags = ['-Wno-unused-parameter',
+                   '-Wno-unused-result',
+                   '-Wno-missing-field-initializers',
+                   '-Wno-sign-compare',
+                   '-Wundef',
+                   '-Wnull-dereference',
+                   '-Wdouble-promotion',
+                   '-Wshadow',
+                   '-Wpointer-arith',
+                   '-Wwrite-strings',
+                   '-Wstrict-overflow=4']
+
+# TODO add clang specific options
+if compiler.get_id() == 'gcc'
+        compiler_cflags += ['-Wformat-signedness',
+                            '-Wduplicated-cond',
+                            '-Wduplicated-branches',
+                            '-Wvla-larger-than=1',
+                            '-Walloc-zero',
+                            '-Wunsafe-loop-optimizations',
+                            '-Wcast-align',
+                            '-Wlogical-op',
+                            '-Wjump-misses-init']
+endif
```

Comment:
I start to hate such lists. What was the reason for selecting -Wfoo but not -Wbar? Can we just set -Wall instead? Or -Wall -Wextra?

Reply:
tbf, I copypasted it from cdba..

Reply:
Should I use `warning_level=everything` from meson instead?

Reply:
I'd prefer to stick to -Wall. And in CDBA too.

Reply:
fwiw https://mesonbuild.com/Builtin-options.html#details-for-warning_level